src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,135 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <stdio.h>
+#include <ctype.h>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#include <stdlib.h>
+
+/* Note: Some systems need both string.h and strings.h (Savage).  However,
+ * some systems can't handle both, only use string.h in that case. */
```

Comment:
I think we can relax formatting rules when migrating code, but it should only take a few minutes if you don't mind.